corehq/form_processor/submission_post.py

```diff
@@ -151,7 +152,16 @@ def _get_success_message(self, instance, cases=None):
         Message is formatted with markdown.
         '''
 
-        if not instance.metadata or instance.metadata.deviceID != FORMPLAYER_DEVICE_ID:
+        if instance.metadata and (not instance.metadata.userID or not instance.metadata.instanceID):
+            metrics_counter('commcare.xform_submissions.partial_metadata', tags={
+                'domain': instance.domain,
+            })
```

Comment:
I think this is right, just double checking. Not returning anything here is intentional, right? Looks like that's emulating previous behavior and is right, but I had a double-take because it makes the branching in this section kind of asymmetrical in a way that's a little hard to think about.

Reply:
yea sorry for the confusion its not supposed to return anything here if there is metadata - there's handlers for that further down the function. But at the same time the submission could satisfy the first statement AND `if instance.metadata.deviceID != FORMPLAYER_DEVICE_ID:` which is why it's separate.

Reply:
Makes sense. Thanks for confirming!